core/communication/bubblewrap/src/main/javascript/index.ts

```diff
@@ -218,49 +230,40 @@ export class Bubblewrap {
 
     /**
      * Constructs a new, independent [[Bubblewrap]] instance with additional registered classes.
+     * There must be a reason for this creating an additional object; unclear to me right now
      *
      * This method throws an exception if there is a duplicate class identifier.
      */
     addClasses(more: Classes): Bubblewrap {
-        const thisKeys = Object.keys(this.classes);
-        const thatKeys = Object.keys(more);
-        for (const thisKey of thisKeys)
-            if (thatKeys.includes(thisKey)) throw new Error(`Duplicate identifier ${thisKey}`);
+        const theseKeys = Object.keys(this.classes);
+        const thoseKeys = Object.keys(more);
+        for (const thatKey of thoseKeys)
+            if (theseKeys.includes(thatKey)) throw new Error(`Duplicate identifier ${thatKey}`);
         return new Bubblewrap({ ...this.classes, ...more }, this.strict);
     }
 
-    private registerStrict(codec: ExtensionCodec): void {
-        if (!this.strict) return;
-
-        const knownPrototypes = [
-            Object.prototype,
-            Error.prototype,
-            Undefined.prototype,
-            ...Object.values(this.classes).map((cls) => cls.prototype),
-        ];
-
-        codec.register({
-            type: msgPackEtypeStrict,
-            encode: (value) => {
-                if (typeof value === "object" && !Array.isArray(value)) {
-                    if (knownPrototypes.includes(Object.getPrototypeOf(value)))
-                        // this value is probably fine, please go on
-                        return null;
-
-                    throw new Error("Attempted to encode an object with an unknown prototype");
-                }
-                return null;
-            },
-            decode: () => {
-                throw new Error("Attempted to decode a dummy type");
-            },
-        });
-    }
-
-    private makeCodec(): ExtensionCodec {
+    private makeAndRegisterCodec(): ExtensionCodec {
         const codec = new ExtensionCodec();
 
-        this.registerStrict(codec);
+        if (this.strict) {
```

Comment:
It's not entirely clear to me why this is being used. Apparently, we're using a specific codec as a guard for not encoding dummy types. However, the `ExtensionCodec` is a private variable which is only called from encode/decode, so it might be enough to just make those checks there.